clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -4607,6 +4606,127 @@ test_unscheduled_initially()
     }
 }
 
+static void
+test_unscheduled_initially_roi()
+{
+#ifdef HAS_ZIP
+    std::cerr
+        << "\n----------------\nTesting initially-unscheduled + time deps with ROI\n";
+    static constexpr int NUM_OUTPUTS = 1;
+    static constexpr memref_tid_t TID_BASE = 100;
+    static constexpr memref_tid_t TID_A = TID_BASE + 0;
+    static constexpr memref_tid_t TID_B = TID_BASE + 1;
+    std::vector<trace_entry_t> refs_A = {
+        make_thread(TID_A),
+        make_pid(1),
+        make_version(TRACE_ENTRY_VERSION),
+        make_timestamp(1001),
+        make_marker(TRACE_MARKER_TYPE_CPU_ID, 0),
+        // A starts out unscheduled but we skip that.
+        make_marker(TRACE_MARKER_TYPE_SYSCALL_UNSCHEDULE, 0),
```

Comment:
Could this really happen in a real trace, an "unschedule" without a later "schedule" somewhere?

If it does, the scheduler would need to wait for the unschedule timeout (that's what we call it right?) right?

Probably orthogonal: What about windowed traces? Each window is a separate trace; but is the scheduler aware that they're from the same run, and try to provide more continuity, and handle missing "unschedule"/"schedule"? Should it?

Reply:
Added a comment:
```
        // (In a real trace some other thread would have to wake up A:
        // we omit that here to keep the test small.)
```

There can be a timeout on an unschedule, but here there is no timeout so the thread would never run unless another thread wakes it up.

Re: windowed traces: probably the same concern would exist for the "simpoints" or any region-of-interest reduction that picks only certain subsets: it does seem possible for the chosen subsets to include an unschedule  but miss its wakeup.  Should we document that somewhere as a caution?  But even without "unschedule" semantics that can happen in any app with condition variables or any synchronization: if you pull out some subsets you could miss wakeups.  I guess most schedulers/simulators wouldn't model those wakeups though and would just run everything.

I added this to `regions_of_interest`:
```
         * Be aware that selecting a subset of code can remove inter-input
         * communication steps that could be required for forward progress.
         * For example, if selected subsets include #TRACE_MARKER_TYPE_SYSCALL_UNSCHEDULE
         * with no timeout but do not include a corresponding
         * #TRACE_MARKER_TYPE_SYSCALL_SCHEDULE for wakeup, an input could remain
         * unscheduled.
```